### PR TITLE
MINOR: Ensure initial topic configs and updates are logged

### PR DESCRIPTION
**Pull request by hachikuji**
This patch adds logging of topic config overrides during creation or during the handling of alter config requests. Also did some minor cleanup to avoid redundant validation logic when adding partitions.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by hachikuji** [comment]
@omkreddy Maybe you can take a look at this?

**Comment by omkreddy** [comment]
retest this please

**Comment by omkreddy** [comment]
retest this please

**Review by omkreddy — APPROVED** [review]
@hachikuji : Thanks for the patch. LGTM

**Comment by omkreddy** [comment]
Java 11 test failures not related, merging to trunk

File: core/src/main/scala/kafka/zk/AdminZkClient.scala [hunk 1]
```diff
@@ -221,15 +201,15 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
...
     }
+
     val proposedAssignment = existingAssignment ++ proposedAssignmentForNewPartitions
     if (!validateOnly) {
       info(s"Creating $partitionsToAdd partitions for '$topic' with the following replica assignment: " +
         s"$proposedAssignmentForNewPartitions.")
-      // add the combined new list
-      createOrUpdateTopicPartitionAssignmentPathInZK(topic, proposedAssignment, update = true)
+
+      writeTopicPartitionAssignment(topic, proposedAssignment, isUpdate = true)
```
Comment (ijuma) [inline core/src/main/scala/kafka/zk/AdminZkClient.scala:210]: I guess we validated that the following checks were redundant?

```scala
     if (partitionReplicaAssignment.values.map(_.size).toSet.size != 1)
       throw new InvalidReplicaAssignmentException("All partitions should have the same number of replicas")
 
     partitionReplicaAssignment.values.foreach(reps =>
       if (reps.size != reps.toSet.size)
        throw new InvalidReplicaAssignmentException("Duplicate replica assignment found: " + partitionReplicaAssignment)
    )
```

Comment (ijuma) [reply core/src/main/scala/kafka/zk/AdminZkClient.scala:210]: Also, `Topic.validate(topic)`.

Comment (hachikuji) [reply core/src/main/scala/kafka/zk/AdminZkClient.scala:210]: @ijuma Yes, these checks are redundant given the checks in `validateReplicaAssignment`, which checks for duplicate replicas as well as verifying the replication factor.

I considered `Topic.validate` as well, but my reasoning was that `addPartitions` depends on topic existence, so the check is not needed. Does that seem reasonable?

Comment (omkreddy) [reply core/src/main/scala/kafka/zk/AdminZkClient.scala:210]: Yes, I have verified above validations while adding partitions.  
and we are checking topic existence before calling addPartitions.

Comment (ijuma) [reply core/src/main/scala/kafka/zk/AdminZkClient.scala:210]: Thanks for the explanation. My concern is that `adminZkClient.addPartitions` is called directly from `TopicCommand` and other methods in `AdminZkClient` generally do their own validation. Also, I'd not be surprised if other people call some of these methods directly. At least ensuring that the topic name is valid (which is cheap), might still be worth it.

File: core/src/main/scala/kafka/zk/AdminZkClient.scala [hunk 2]
```diff
@@ -140,20 +125,16 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
         throw new InvalidReplicaAssignmentException("Duplicate replica assignment found: " + partitionReplicaAssignment)
```
Comment (sumannewton) [inline core/src/main/scala/kafka/zk/AdminZkClient.scala]: Shouldn't there be a check to validate if all replicas passed are existing and available brokers during topic creation? 
PR #5665 has a check to guarantee at least one replica is alive during topic creation. Just take a look. 
Also, shouldn't we refactor and maintain uniformity in [ReassignPartitionsCommand](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala#L316) validations and [AdminZkClient](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/zk/AdminZkClient.scala#L120) validations?

File: core/src/main/scala/kafka/zk/AdminZkClient.scala [hunk 3]
```diff
@@ -221,15 +201,15 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
...
-      // add the combined new list
-      createOrUpdateTopicPartitionAssignmentPathInZK(topic, proposedAssignment, update = true)
+
+      writeTopicPartitionAssignment(topic, proposedAssignment, isUpdate = true)
     }
     proposedAssignment
-
   }
 
   private def validateReplicaAssignment(replicaAssignment: Map[Int, Seq[Int]],
```
Comment (sumannewton) [inline core/src/main/scala/kafka/zk/AdminZkClient.scala:215]: Shouldn't there be a check to validate if all replicas passed are existing and available brokers during topic creation? 
PR #5665 has a check to guarantee at least one replica is alive during topic creation. Just take a look. It also does some refactoring.
Also, shouldn't we refactor and maintain uniformity in [ReassignPartitionsCommand](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala#L316) validations and [AdminZkClient](https://github.com/apache/kafka/blob/928f45f61f793f0b2d7df11ff92463d7edda60d6/core/src/main/scala/kafka/zk/AdminZkClient.scala#L120) validations?

**Pull request closed without merging**